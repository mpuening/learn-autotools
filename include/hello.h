#ifndef HELLO_H_
#define HELLO_H_

int hello(void);

#endif
