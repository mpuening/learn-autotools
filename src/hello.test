#!/bin/bash

echo 1..2 # Test plan; 2 tests

HELLO=./hello

function test_build {
	if [ -f "$HELLO" ]; then
		echo "ok build - $HELLO executable found. # OK"
	else 
		echo "not ok build - $HELLO executable not found # FAIL"
		return
	fi
}

function test_output {
	$HELLO > hello.out
	diff hello.expected hello.out

	if [ $? -eq 0 ]; then
		echo "ok output - correct hello output # OK"
	else
		echo "not ok output - incorrect hello output # FAIL"		
	fi
	rm hello.out
}

test_build
test_output

exit 0



# How to report on test cases

# https://www.gnu.org/software/automake/manual/html_node/Use-TAP-with-the-Automake-test-harness.html

#echo 1..3 # Test plan; 3 tests
#echo "ok 1 - Output equals # OK"
#echo "not ok 2 - Test failed # FAIL"
#echo "ok 3 - Test skipped # SKIP"
