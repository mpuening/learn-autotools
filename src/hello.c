#include <stdio.h>

#include "../include/hello.h"

int hello(void) {
	puts("Hello World, Y'all.");
	return 0;
}
