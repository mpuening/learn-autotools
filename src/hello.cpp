#include <iostream>

#include "../include/hello.h"

using namespace HELLO;
using namespace std;

void Hello::runApplication()
{
    cout << "Hello World, Y'all! (In C++)" << endl;
}
