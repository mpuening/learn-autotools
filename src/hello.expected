Running hello 1.0.
Util version: 0.0.1-SHAPSHOT
Hello World, Y'all! (In C++)
